contracts/utils/SignatureValidator.sol

```diff
@@ -99,4 +100,38 @@ contract SignatureValidator {
 
     return signer;
   }
+
+ /**
+   * @notice Returns true if the provided signature is valid for the given signer.
+   * @dev Supports SignatureType.EIP712, SignatureType.EthSign, and ERC1271 signatures
+   * @param _hash      Hash that was signed
+   * @param _signer    Address of the signer candidate
+   * @param _signature Signature byte array
+   */
+  function isValidSignature(
+    bytes32 _hash,
+    address _signer,
+    bytes memory _signature
+  ) internal view returns (bool valid) {
+    uint256 signatureType = uint8(_signature[_signature.length - 1]);
+
+    if (signatureType == SIG_TYPE_EIP712 || signatureType == SIG_TYPE_ETH_SIGN) {
+      // Recover signer and compare with provided
+      valid = recoverSigner(_hash, _signature) == _signer;
```

Comment:
The `recoverSigner` really expects a 66-bytes array but can accept and process larger arrays. And now it looks like it is actually possible to pass this larger array to the function. I can't find any malicious use case for that.

Reply:
Thanks, good catch! This makes signatures mutable so it's better to fix it